### ISSUE TITLE
Median/quantile/bottoms*/tops* can be log(n) faster

### ISSUE THREAD

**Issue by mourner** [observablehq/array#1]
Median/quantile/bottoms*/tops* fully sort the data prior to returning the results, which is `O(n log n)`. However, all those could be `O(n)` by using a [selection algorithm](https://en.wikipedia.org/wiki/Selection_algorithm) instead. 

For an example implementation, see https://github.com/simple-statistics/simple-statistics/pull/146
I also have a fast standalone selection module: https://github.com/mourner/quickselect

**Comment by mbostock** [comment]
Of course; I mentioned selection algorithms in the linked notebook. Your contribution would be welcome here! And if not, I’ll do it when I find time. Thanks for the reminder.
